src/fingerprinting/hooks.ts

```diff
@@ -0,0 +1,87 @@
+import { FingerprintInjector } from 'fingerprint-injector';
+import { BrowserPool, PlaywrightPlugin, PuppeteerPlugin } from '..';
+import { BrowserController } from '../abstract-classes/browser-controller';
+import { LaunchContext } from '../launch-context';
+import { getGeneratorDefaultOptions } from './utils';
+
+export const createFingerprintPreLaunchHook = (browserPool: BrowserPool<any, any, any, any, any>) => {
```

Comment:
in general i would prefer named functions instead of arrow function variables. they are much easier to type (e.g. add return type) and feels more readable. also looks better in the generated API docs we have in the SDK.